nil/internal/execution/block_generator.go

```diff
@@ -102,27 +95,48 @@ func (g *BlockGenerator) Rollback() {
 	g.rwTx.Rollback()
 }
 
-func (p *BlockGenerator) CollectGasPrices(prevBlockHash common.Hash, shardHashes []common.Hash) []types.Uint256 {
+func (p *BlockGenerator) CollectGasPrices(prevBlockId types.BlockNumber) []types.Uint256 {
 	if !p.params.ShardId.IsMainShard() {
 		return nil
 	}
 
+	// Basically we load configuration from block.MainChainHash.
+	// But for main shard this value should be block.PrevBlock.
+	// The first block uses configuration from itself.
+	configBlockId := prevBlockId
+	if prevBlockId != 0 {
```

Comment:
Wouldn't this condition also trigger for block zero on regular shards?

Reply:
No. Because we have on the function top.
```go
        if !p.params.ShardId.IsMainShard() {
		return nil
	}
```